ludwig/schema/preprocessing.py

```diff
@@ -0,0 +1,751 @@
+from abc import ABC
+from dataclasses import field
+from typing import Any, ClassVar, Optional, Union
+
+from marshmallow import fields, ValidationError
+from marshmallow_dataclass import dataclass
+
+from ludwig.schema import utils as schema_utils
+from ludwig.utils import strings_utils
+from ludwig.utils.registry import Registry
+from ludwig.utils.tokenizers import tokenizer_registry
+
+preprocessing_registry = Registry()
+
+
+def register_preprocessor(name: str):
+    def wrap(preprocessing_config: BasePreprocessingConfig):
+        preprocessing_registry[name] = preprocessing_config
+        return preprocessing_config
+
+    return wrap
+
+
+@dataclass
+class BasePreprocessingConfig(schema_utils.BaseMarshmallowConfig, ABC):
+    """Base class for input feature preprocessing. Not meant to be used directly.
+
+    The dataclass format prevents arbitrary properties from being set. Consequently, in child classes, all properties
+    from the corresponding input feature class are copied over: check each class to check which attributes are different
+    from the preprocessing of each feature.
+    """
+
+    feature_type: ClassVar[Optional[str]] = None
+    "Class variable pointing to the corresponding preprocessor."
+
+    type: str
+
+
+@dataclass
+@register_preprocessor("text")
+class TextPreprocessingConfig(schema_utils.BaseMarshmallowConfig):
+    """TextPreprocessingConfig is a dataclass that configures the parameters used for a text input feature."""
+
+    pretrained_model_name_or_path: Optional[str] = schema_utils.String(
+        default=None,
+        allow_none=True,
+        description="This can be either the name of a pretrained HuggingFace model or a path where it was downloaded",
+    )
+
+    tokenizer: Optional[str] = schema_utils.String(
+        default="space_punct",
```

Comment:
Could we restrict the options to `tokenizer_registry.keys()`?

Reply:
Same as above, will include in the encoder schema refactor.

Reply:
Done!